crates/storage/db/benches/criterion.rs

```diff
@@ -0,0 +1,155 @@
+    measure_table_db::<CumulativeTxCount>(&mut group);
+    measure_table_db::<NonCanonicalTransactions>(&mut group);
+    measure_table_db::<Transactions>(&mut group);
+    measure_table_db::<TxHashNumber>(&mut group);
+    measure_table_db::<Receipts>(&mut group);
+    measure_table_db::<Logs>(&mut group);
+    measure_table_db::<PlainAccountState>(&mut group);
+    measure_table_db::<PlainStorageState>(&mut group);
+    measure_table_db::<Bytecodes>(&mut group);
+    measure_table_db::<AccountHistory>(&mut group);
+    measure_table_db::<StorageHistory>(&mut group);
+    measure_table_db::<AccountChangeSet>(&mut group);
+    measure_table_db::<StorageChangeSet>(&mut group);
+    measure_table_db::<TxSenders>(&mut group);
+    measure_table_db::<Config>(&mut group);
+    measure_table_db::<SyncStage>(&mut group);
+}
+
+pub fn serialization(c: &mut Criterion) {
+    let mut group = c.benchmark_group("tables_serialization");
+    group.measurement_time(std::time::Duration::from_millis(100));
+    group.warm_up_time(std::time::Duration::from_millis(100));
+
+    measure_table_serialization::<CanonicalHeaders>(&mut group);
+    measure_table_serialization::<HeaderTD>(&mut group);
+    measure_table_serialization::<HeaderNumbers>(&mut group);
+    measure_table_serialization::<Headers>(&mut group);
+    measure_table_serialization::<BlockOmmers>(&mut group);
+    measure_table_serialization::<CumulativeTxCount>(&mut group);
+    measure_table_serialization::<NonCanonicalTransactions>(&mut group);
+    measure_table_serialization::<Transactions>(&mut group);
+    measure_table_serialization::<TxHashNumber>(&mut group);
+    measure_table_serialization::<Receipts>(&mut group);
+    measure_table_serialization::<Logs>(&mut group);
+    measure_table_serialization::<PlainAccountState>(&mut group);
+    measure_table_serialization::<PlainStorageState>(&mut group);
+    measure_table_serialization::<Bytecodes>(&mut group);
+    measure_table_serialization::<AccountHistory>(&mut group);
+    measure_table_serialization::<StorageHistory>(&mut group);
+    measure_table_serialization::<AccountChangeSet>(&mut group);
+    measure_table_serialization::<StorageChangeSet>(&mut group);
+    measure_table_serialization::<TxSenders>(&mut group);
+    measure_table_serialization::<Config>(&mut group);
+    measure_table_serialization::<SyncStage>(&mut group);
+}
+
+fn measure_table_serialization<T>(group: &mut BenchmarkGroup<WallTime>)
+where
+    T: Table + Default,
+    T::Key: Default + Clone,
+    T::Value: Default + Clone,
+{
+    let pair = &load_vectors::<T>();
+    group.bench_function(format!("{}.KeyEncode", T::NAME), |b| {
+        b.iter(|| {
+            for (k, _, _, _) in pair {
+                k.clone().encode();
+            }
+        })
+    });
+
+    group.bench_function(format!("{}.KeyDecode", T::NAME), |b| {
+        b.iter(|| {
+            for (_, k, _, _) in pair {
+                let _ = <T as Table>::Key::decode(k.clone());
```

Comment:
May need black_box? Given its not used anywhere compiler may elide it. Same comment as above re clone